sql/src/test/java/io/crate/operation/collect/LuceneDocCollectorTest.java

```diff
@@ -403,4 +411,23 @@ public void testOrderByFieldVisitorExpressions() throws Exception {
                 "{\"continent\":\"America\",\"countryName\":\"USA\",\"population\":1000}| 1000\n" +
                 "{\"continent\":\"America\",\"countryName\":\"USA\",\"population\":1001}| 1001\n"));
     }
+
+    @Test
+    public void testCircuitBreak() throws Exception {
+        CollectingRowReceiver projector = CollectingRowReceiver.withPauseAfter(5);
+        RamAccountingContext ramAccountingContext = mock(RamAccountingContext.class);
+        CrateCollector docCollector = createDocCollector("select population from countries limit 15", projector, ramAccountingContext);
+
+        when(ramAccountingContext.trippedBreaker()).thenReturn(false);
```

Comment:
it is not needed to first let 5 docs collect and only then circuit break, it can just break immediately